frontend/src/metabase/components/AccordionList.jsx

```diff
@@ -247,19 +268,29 @@ export default class AccordionList extends Component {
     const sectionIsTogglable = sectionIndex =>
       alwaysTogglable || sections.length > 1;
 
-    let { searchText } = this.state;
+    const { searchText } = this.state;
     let searchFilter = () => true;
     if (searchText) {
       searchFilter = item => {
-        let itemText = String(item[searchProp] || "");
-        if (searchCaseInsensitive) {
-          itemText = itemText.toLowerCase();
-          searchText = searchText.toLowerCase();
-        }
-        if (searchFuzzy) {
-          return itemText.indexOf(searchText) >= 0;
-        } else {
-          return itemText.startsWith(searchText);
+        if (typeof searchProp === "string") {
+          return this.searchPred(
+            item,
+            searchText,
+            searchProp,
+            searchCaseInsensitive,
+            searchFuzzy,
+          );
+        } else if (searchProp.constructor === Array) {
+          const searchResults = searchProp.map(member =>
+            this.searchPred(
+              item,
+              searchText,
+              member,
+              searchCaseInsensitive,
+              searchFuzzy,
+            ),
+          );
+          return searchResults.reduce((acc, curr) => acc || curr);
```

Comment:
`searchProp`, `searchCaseInsensitive` and `searchFuzzy` are props and `searchText` is state. Should we maybe extract them right in the `searchPred` method to keep this block more compact? I mean:

```js
searchPred = item => {
   const {searchProp, searchCaseInsensitive, searchFuzzy} = this.props;
   const {searchText} = this.state;
   // actual implementation
}
```

Reply:
need to pass in searchprop but otherwise sgtm